### PR TITLE
Perform datagen for the new symbols keys

### PR DESCRIPTION
**Pull request by Manishearth**
Part of https://github.com/unicode-org/icu4x/issues/3865

Based on https://github.com/unicode-org/icu4x/pull/4199

### CODE REVIEW

**Comment by sffc** [comment]
There are 1145 entries in the locales list, 435 unique data structs, and many more locales that are being stripped by the runtime provider because they are reachable by fallback. Looks to me like deduplication is doing its job.

It appears that all keys would fit into an 8-bit VZV index array. The longest struct is `ML_X_5S` which is about 211 bytes long. It appears that language uses 3-bit UTF-8. ff-Adlm is I think around the same length with 4-bit UTF-8. So we're not guaranteed to never hit the limit but at least with today's CLDR data it appears that we fit. This change would save about 3KB total.

**Review by sffc — APPROVED** [review]

File: provider/datagen/src/transform/cldr/datetime/neo.rs [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+
+    // Tostring can be removed when we delete symbols.rs, or we can perhaps refactor it to use Value
+    let calendar_str = calendar.to_string();
+    let (map, has_leap) = super::symbols::get_month_code_map(&calendar_str);
+
+    if has_leap {
+        let mut symbols: BTreeMap<UnvalidatedTinyAsciiStr<4>, &str> = BTreeMap::new();
+        for (k, v) in months.0.iter() {
+            let code = if k == "7-yeartype-leap" && *calendar == value!("hebrew") {
+                tinystr!(4, "M06L")
```
Comment (sffc) [inline provider/datagen/src/transform/cldr/datetime/neo.rs:250]: Nit before merge: make sure this generates the right table for Hebrew. We spent a while getting this right with Pedro.

**Review by sffc — APPROVED** [review]

**Comment by Manishearth** [comment]
https://github.com/unicode-org/icu4x/compare/c26d9eab0177b4ea2b32bbc55bf2f3bfd82c4b7f..6b307bd7b336105b3401311f4e8b3fca4b2e7611

**Review by sffc — APPROVED** [review]

**Comment by Manishearth** [comment]
autosubmit=on

**Comment by Manishearth** [comment]
CI passes, ready for a stamp!

**Review by sffc — APPROVED** [review]

**Pull request merged**